src/main/java/com/o19s/es/ltr/utils/Scripting.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright [2017] Wikimedia Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.o19s.es.ltr.utils;
+
+import org.apache.lucene.expressions.Expression;
+import org.apache.lucene.expressions.js.JavascriptCompiler;
+import org.elasticsearch.SpecialPermission;
+import org.elasticsearch.script.ClassPermission;
+import org.elasticsearch.script.ScriptException;
+
+import java.security.AccessControlContext;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Scripting {
+    private Scripting() {}
+
+    public static Object compile(String scriptSource) {
+        // classloader created here
+        final SecurityManager sm = System.getSecurityManager();
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
+        }
+        return AccessController.doPrivileged(new PrivilegedAction<Expression>() {
+            @Override
+            public Expression run() {
+                try {
+                    // snapshot our context here, we check on behalf of the expression
+                    AccessControlContext engineContext = AccessController.getContext();
+                    ClassLoader loader = getClass().getClassLoader();
```

Comment:
It looks like getting the class loader should be extracted in its own method, and that `getClass().getClassLoader()` should be guarded by an `if (sm == null)`. Unless we do expect an exception on that call... in which case, we probably want a few comments explaining what is going on.